app/src/main/java/org/p2p/wallet/swap/ui/jupiter/routes/SwapSelectRoutesBottomSheet.kt

```diff
@@ -0,0 +1,95 @@
+package org.p2p.wallet.swap.ui.jupiter.routes
+
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.lifecycleScope
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.koin.android.ext.android.inject
+import org.p2p.uikit.components.finance_block.FinanceBlockCellModel
+import org.p2p.uikit.components.finance_block.UiKitFinanceBlockView
+import org.p2p.uikit.components.finance_block.baseCellDelegate
+import org.p2p.uikit.model.AnyCellItem
+import org.p2p.uikit.utils.attachAdapter
+import org.p2p.wallet.R
+import org.p2p.wallet.common.ui.bottomsheet.BaseBottomSheet
+import org.p2p.wallet.databinding.DialogSwapSelectRoutesBinding
+import org.p2p.wallet.home.ui.main.bottomsheet.HomeActionsBottomSheet
+import org.p2p.wallet.swap.jupiter.repository.model.JupiterSwapRoute
+import org.p2p.wallet.swap.jupiter.statemanager.SwapState
+import org.p2p.wallet.swap.jupiter.statemanager.SwapStateAction
+import org.p2p.wallet.swap.jupiter.statemanager.SwapStateManager
+import org.p2p.wallet.swap.jupiter.statemanager.SwapStateManagerHolder
+import org.p2p.wallet.swap.ui.jupiter.settings.adapter.SwapSettingsAdapter
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+private const val ARG_STATE_MANAGE_KEY = "ARG_STATE_MANAGE_KEY"
+
+class SwapSelectRoutesBottomSheet : BaseBottomSheet() {
+
+    companion object {
+        fun show(
+            fm: FragmentManager,
+            stateManagerKey: String
+        ) {
+            val tag = HomeActionsBottomSheet::javaClass.name
+            if (fm.findFragmentByTag(tag) != null) return
+            SwapSelectRoutesBottomSheet()
+                .withArgs(ARG_STATE_MANAGE_KEY to stateManagerKey)
+                .show(fm, tag)
+        }
+    }
+
+    private val binding: DialogSwapSelectRoutesBinding by viewBinding()
+    private val stateManagerKey: String by args(ARG_STATE_MANAGE_KEY)
+    private val managerHolder: SwapStateManagerHolder by inject()
+    private val mapper: SwapSelectRoutesMapper by inject()
+    private val stateManager: SwapStateManager
+        get() = managerHolder.get(stateManagerKey)
+
+    private val adapter = SwapSettingsAdapter(
+        baseCellDelegate(inflateListener = { financeBlock ->
+            financeBlock.setOnClickAction { view, item -> onRouteClick(item, view) }
+        }),
+    )
+
+    override fun getTheme(): Int = R.style.WalletTheme_BottomSheet_Rounded
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.dialog_swap_select_routes, container, false)
```

Comment:
i think by viewBinding already inflates the layout, no?

Reply:
no, we need to pass it